### ISSUE TITLE
fullydistributed triangulation made from description incl. manifolds results in disjoint faces across at subdomain borders

### ISSUE THREAD

**Issue by richardschu** [dealii/dealii#12665]
Hi everyone,
I might have found a bug in parallel::fullydistributed::Triangulation (pfdt) when I was trying to create a pfdt from a serial triangulation, although it might also be rooted in wrong use of the functions.
My testcase is a circular disk in 2D with a circular inclusion which I read from an .msh file (a similar mesh can be created combining a `hyper_shell` and a `hyper_ball`):
![circ_inclusion](https://user-images.githubusercontent.com/69867655/129874100-623b8331-8c92-4a1e-91a0-66dd9fd36ad4.png)
Then I do the following:
1) Read in a grid with `read_msh()` into a serial triangulation.
2) In cells touching the boundary `cell->set_all_manifold_ids(100)`
3) Attach a `SphericalManifold` to both the serial triangulation and pfdt (similar as in `tests/fullydistributed_grids/copy_serial_tria_03.cc`)
4) Refine serial triangulation via `refine_global()`
5) Partition serial triangulation via `partition_triangulation`
6) Create description of the serial triangulation via `create_description_from_triangulation`
7) Actually create pfdt via `create_triangulation()`
If I do so, the created pfdt features disjoint cell faces across subdomains, as can be seen here:
![circ_inclusion_3_procs](https://user-images.githubusercontent.com/69867655/129875564-3965dce1-bdd3-4da8-bf12-8a6710150ebe.png)
I can also replace 6) and 7) by `copy_triangulation()`, which gives the expected results, so, no disjoint cells, but also has a completely different partitioning.
But am I using above mentioned function(s) in a wrong way?
Any comments, suggestions, etc. are highly welcome!
My MWE plus the mesh file (in dummy .txt file format):
[circ_inclusion.txt](https://github.com/dealii/dealii/files/7006226/circ_inclusion.txt)
[copy_serial_tria_03.txt](https://github.com/dealii/dealii/files/7006227/copy_serial_tria_03.txt)

EDIT:
If one does not attach a manifold at all, 6) and 7) do work as expected and in `tests/fullydistributed_grids/copy_serial_tria_03.cc` all cells do have the same manifold attached to them. So it seems, that a cell does not know, which manifolds are attached on other processors to map the mesh in a conforming way (its assuming FlatManifold?) - which sounds reasonable! I guess this functionality is simply impossible due to the design of pfdt?

**Comment by peterrum** [comment]
I couldn't find the bug right away. But the bright side is that if the mesh is created with the multigrid levels, the problem does not happen:

```diff
diff --git a/main.cc b/main.cc
index f19affb..9698421 100644
--- a/main.cc
+++ b/main.cc
@@ -45,7 +45,7 @@ test
    MPI_Comm comm)
 {
   // create triangulations
-  Triangulation<dim> base_tria;
+  Triangulation<dim> base_tria(Triangulation<dim>::limit_level_difference_at_vertices);
   parallel::fullydistributed::Triangulation<dim> tria_pft(comm);
 
   // Read in the basic mesh from a file.
@@ -78,6 +78,7 @@ test
     base_tria.refine_global (n_glob_ref);
 
   GridTools::partition_triangulation(Utilities::MPI::n_mpi_processes(comm), base_tria);
+  GridTools::partition_multigrid_levels(base_tria);
   
   if (construct_from_data == 1)
     {
@@ -85,7 +86,7 @@ test
          auto construction_data =
                TriangulationDescription::Utilities::create_description_from_triangulation(
                  base_tria,
-                 comm);
+                 comm, TriangulationDescription::Settings::construct_multigrid_hierarchy);
          tria_pft.create_triangulation(construction_data);
     }
   else if (construct_from_data == 0)

```

My best guess is that the manifold id of the parent is not communicated. I'll continue debugging later today!

**Comment by kronbichler** [comment]
Doesn't this smell a bit like a bug? I assumed we treat volumes, interior surfaces and boundaries similarly, so we should do the same here. And just for me to understand, the problem is that we have a *coarse* grid cell whose face happens to be on the boundary on one rank but not the other?

**Comment by peterrum** [comment]
> And just for me to understand, the problem is that we have a coarse grid cell whose face happens to be on the boundary on one rank but not the other?

Exactly. This is just the problem in the case of p:f:T, since in the other trias have the whole coarse-grid tria.

**Comment by kronbichler** [comment]
I'm not sure #12669 is the right solution. The more I look at the code, the more I believe this is simply a left-over from when we did not have manifolds but only `Boundary` objects. In my opinion we should treat interior cells and cells at the boundary exactly the same. Hence, I believe the different cases should be removed. I anticipate some failures in the tests, but this really isn't a good solution right now.

**Comment by richardschu** [comment]
Hi @peterrum and @kronbichler - thank for taking a look!
Well actually that 'hack' you suggested as a temporary fix was actually the first thing I tried given it was in the original file.
However, it does not do the trick for me ... so I compiled the most recent master and still it gives me the same effects. Did you run the example on your machine and really got decent results without 'holes'?
Also, I did not tell you is that the outer cells also have a material_id() == 1, allowing for
```
if (cell->material_id() == 0)
  cell->set_all_boundary_ids(1);
```
which gives exactly the same result as explained before.
I do not know, if that actually matters, but anyways ...
